Gemfile

```diff
@@ -31,6 +31,7 @@ gem "pry-rails", group: :development
 gem "puma", "~> 4.1"
 gem "rails", "~> 6.0.3", ">= 6.0.3.2"
 gem "rails-i18n"
+gem "redis"
```

Comment:
gem này có tác dụng gì e

Reply:
- Redis là data structures server.
- Nó cung cấp quyền truy cập dữ liệu thông qua một tập các câu lệnh, các request sử dụng cấu trúc server-client với giao thức TCP sockets và một giao thức đơn giản khác. 
=> Vì vậy, các tiến trình khác nhau có thể query hay modify cùng một dữ liệu dưới nhiều cách khác nhau.
- là một in-memory data structure store, điều này có nghĩa là Redis lưu dữ liệu ở trong bộ nhớ chính (RAM)
=> Tốc độ nhanh